Sources/TuistKit/Lint/CodeLinter.swift

```diff
@@ -0,0 +1,63 @@
+import Foundation
+import TSCBasic
+import TuistCore
+import TuistSupport
+
+protocol CodeLinting {
+    func lint(sources: AbsolutePath, path: AbsolutePath) throws
+}
+
+class CodeLinter: CodeLinting {
+    private let rootDirectoryLocator: RootDirectoryLocating
+    private let binaryLocator: BinaryLocating
+
+    init(rootDirectoryLocator: RootDirectoryLocating = RootDirectoryLocator(),
+         binaryLocator: BinaryLocating = BinaryLocator())
+    {
+        self.rootDirectoryLocator = rootDirectoryLocator
+        self.binaryLocator = binaryLocator
+    }
+
+    // MARK: - CodeLinting
+
+    func lint(sources: AbsolutePath, path: AbsolutePath) throws {
+        let swiftLintPath = try binaryLocator.swiftLintPath()
+        let swiftLintConfigPath = self.swiftLintConfigPath(path: path)
+        let swiftLintArguments = buildSwiftLintArguments(swiftLintPath: swiftLintPath,
+                                                         sources: sources,
+                                                         configPath: swiftLintConfigPath)
+
+        let result = try System.shared.observable(swiftLintArguments)
+            .mapToString()
+            .toBlocking()
+            .toArray()
+
+        // TODO: swiftlint errors should break tuist (?)
+        result
+            .forEach {
+                logger.notice(Logger.Message(stringLiteral: $0.value))
+            }
```

Comment:
Can we deal with errors better?

Reply:
Have you considered using [this](https://github.com/tuist/tuist/blob/master/Sources/TuistSupport/System/Observable+System.swift#L14). It prints the messages forwared through the standard error automatically.